java-checks-testkit/src/test/java/org/sonar/java/checks/verifier/TestUtils.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.verifier;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.internal.TestInputFileBuilder;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class TestUtils {
```

Comment:
Unfortunately, after adding this `TestUtils` class we will have 4 identical `TestUtils` classes duplicated:
```
java-frontend/src/test/java/org/sonar/java/TestUtils.java
java-checks-testkit/src/test/java/org/sonar/java/checks/verifier/TestUtils.java
java-symbolic-execution/src/test/java/org/sonar/java/se/utils/TestUtils.java
java-checks/src/test/java/org/sonar/java/CheckTestUtils.java
```
Every module depends on `java-frontend` but it's not possible to use the test code of this module in others.
For `java-checks-testkit` it's a different story, we can move `TestUtils` from `src/test/java` to `src/main/java` so we can use it in `java-symbolic-execution` and `java-checks`.
Moving `TestUtils`, require `junit-jupiter` dependency just for one line:
```
assertTrue(file.exists(), "Path '" + path + "' should exist.");
```
We could replace this line with:
```
    if (!file.exists()) {
      throw new IllegalStateException("Path '" + path + "' should exist.");
    }
```
And once we have `java-checks-testkit/src/main/java/org/sonar/java/checks/verifier/TestUtils.java` we can remove 2 of the `TestUtils` classes:
```
java-checks/src/test/java/org/sonar/java/CheckTestUtils.java
java-symbolic-execution/src/test/java/org/sonar/java/se/utils/TestUtils.java
java-symbolic-execution/src/test/java/org/sonar/java/se/utils/TestUtilsTest.java
```
By renaming in all files:
`org.sonar.java.se.utils.TestUtils` => `org.sonar.java.checks.verifier.TestUtils`
`org.sonar.java.CheckTestUtils` => `org.sonar.java.checks.verifier.TestUtils`
`CheckTestUtils.testSourcesPath(` => `TestUtils.testSourcesPath(`
`CheckTestUtils.inputFile(` => `TestUtils.inputFile(`